### PR TITLE
staticanalysis/bot: Run clang-format on each file from the patch

### LINKED ISSUE #1815: staticanalysis/bot: clang-format reformats code that is correctly formatted
The first thing that the bot does is to performa a ```./mach clang-format```[[1]](https://github.com/abpostelnicu/release-services/blob/master/src%2Fstaticanalysis%2Fbot%2Fstatic_analysis_bot%2Fclang%2Fformat.py#L43), that translates to running clang-format on a diff[[2]](https://searchfox.org/mozilla-central/source/python/mozbuild/mozbuild/mach_commands.py#2749). 
In most cases this is fine but there are some scenarios where context is needed by clang-format to better determine the token types that need to be reformat and thus errors happen, please see [[3]](https://phabricator.services.mozilla.com/D16388) and [[4]](https://bugzilla.mozilla.org/show_bug.cgi?id=1513626).
Since we already formatted the entire repo I think it's safe to assume that we want to do a ```per file``` basis format where we call:
```./mach clang-format -p file```
for all of the eligible files from the patch.

### PR DESCRIPTION
**Pull request by abpostelnicu**
The purpose of this PR is to run clang-format of every file from the patch instead of running it on the diff.

For more details on why we want this behaviour please see https://github.com/mozilla/release-services/issues/1815

Fixes #1815.

### CODE REVIEW

File: src/staticanalysis/bot/static_analysis_bot/clang/format.py [hunk 1]
```diff
@@ -35,19 +35,32 @@ def __init__(self):
...
             'gecko-env',
-            './mach', '--log-no-times', 'clang-format',
+            './mach', '--log-no-times', 'clang-format', '-p'
         ]
-        logger.info('Running ./mach clang-format', cmd=' '.join(cmd))
 
-        # Run command
-        clang_output = subprocess.check_output(cmd, cwd=settings.repo_dir).decode('utf-8')
+        # Append to the cmd list each C/C++ file for format
+        found_eligible_files = False
```
Comment (sylvestre) [inline src/staticanalysis/bot/static_analysis_bot/clang/format.py]: what about moving that in a function instead?

Comment (abpostelnicu) [reply src/staticanalysis/bot/static_analysis_bot/clang/format.py]: Sure I have no objections against that.

Comment (abpostelnicu) [reply src/staticanalysis/bot/static_analysis_bot/clang/format.py]: Updated with latest modifications that adds a function which returns a list of files to be sent to format.

File: src/staticanalysis/bot/static_analysis_bot/clang/format.py [hunk 2]
```diff
@@ -35,19 +35,39 @@ def __init__(self):
...
         ]
-        logger.info('Running ./mach clang-format', cmd=' '.join(cmd))
 
-        # Run command
-        clang_output = subprocess.check_output(cmd, cwd=settings.repo_dir).decode('utf-8')
+        # Returns a list of eligible files for format
+        def get_eligible_files():
+            files = []
+            # Append to the files list each C/C++ file for format
+            for file in revision.files:
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/clang/format.py]: Nit: don't use `file` as it's a reserved keyword

File: src/staticanalysis/bot/static_analysis_bot/clang/format.py [hunk 3]
```diff
@@ -35,19 +35,39 @@ def __init__(self):
...
             'gecko-env',
-            './mach', '--log-no-times', 'clang-format',
+            './mach', '--log-no-times', 'clang-format', '-p'
         ]
-        logger.info('Running ./mach clang-format', cmd=' '.join(cmd))
 
-        # Run command
-        clang_output = subprocess.check_output(cmd, cwd=settings.repo_dir).decode('utf-8')
+        # Returns a list of eligible files for format
+        def get_eligible_files():
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/clang/format.py]: Nit: you could use a list comprehesion, more Pythonic:
```Python
files_to_format = [f for f in revision.files if os.path.splitext(file)[1].lower() in settings.cpp_extensions]
```

File: src/staticanalysis/bot/static_analysis_bot/clang/format.py [hunk 4]
```diff
@@ -35,19 +35,39 @@ def __init__(self):
...
+        if not files_to_format:
+            logger.info('No eligible files found to format.')
+            return []
+
+        # Append to the cmd the files that will be formatted
+        cmd += files_to_format
+
+        # Run command and commit the current revision for `./mach clang-format ...` to reformat its changes
+        logger.info('Running ./mach clang-format', cmd=' '.join(cmd))
+        clang_output = subprocess.check_output(
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/clang/format.py]: Nit: this could be on one line, the char limit for one line is pretty high (something like ~150 chars)

File: src/staticanalysis/bot/static_analysis_bot/clang/format.py [hunk 5]
```diff
@@ -63,6 +83,7 @@ def run(self, revision):
             os.path.join(settings.repo_dir, path).encode('utf-8')  # needed for hglib
             for path in filter(settings.is_allowed_path, revision.files)
         ]
+
```
Comment (marco-c) [inline src/staticanalysis/bot/static_analysis_bot/clang/format.py]: Nit: unrelated whitespace change

**Comment by marco-c** [comment]
>  Should we exclude third-party files?

Or are they excluded by mach directly?

File: src/staticanalysis/bot/static_analysis_bot/clang/format.py (hunk 5 above)
Comment (abpostelnicu) [reply src/staticanalysis/bot/static_analysis_bot/clang/format.py]: I added the whitespace in order to better separated, I don't want to mix the contexts.

**Comment by abpostelnicu** [comment]
> > Should we exclude third-party files?
> 
> Or are they excluded by mach directly?

It's handled by ```mach```[1](https://searchfox.org/mozilla-central/source/python/mozbuild/mozbuild/mach_commands.py#2773).

File: src/staticanalysis/bot/static_analysis_bot/clang/format.py (hunk 5 above)
Comment (marco-c) [reply src/staticanalysis/bot/static_analysis_bot/clang/format.py]: I like it too, it's just that it's unrelated to the current patch (anyway it's a "nit", you can ignore it).
